ddtrace/tracer/writer.go

```diff
@@ -219,6 +219,22 @@ func (h *logTraceWriter) encodeSpan(s *span) {
 		h.buf.WriteString(":")
 		h.marshalString(v)
 	}
+	// We cannot pack messagepack into JSON, so we need to marshal the meta struct as JSON, and send them through the `meta` field
+	for k, v := range s.MetaStruct {
+		if first {
+			first = false
+		} else {
+			h.buf.WriteString(`,`)
+		}
+		h.marshalString(k)
+		h.buf.WriteString(":")
+		jsonValue, err := json.Marshal(v)
+		if err != nil {
+			log.Error("Error marshaling value %q: %v", v, err)
+			continue
+		}
+		h.marshalString(string(jsonValue))
```

Comment:
Is there no way to avoid copying to the string here? It's silly because it'll inevitably end up in a `[]byte` again at the end...

Reply:
This writer is the debug writer basically so I don't think introducting unsafe stuff like we did in go-libddwaf is not ideal

Reply:
Well, It was used to report traces in lambda years ago but it's been deprecated for a long time now

Reply:
Oh okay